tests/query/processors/test_mandatory_condition_applier.py

```diff
@@ -0,0 +1,80 @@
+import pytest
+from typing import List
+
+# from snuba.clickhouse.columns import ColumnSet, String
+
+from snuba.datasets.schemas.tables import TableSource
+from snuba.query.logical import Query
+from snuba.query.processors.mandatory_condition_applier import MandatoryConditionApplier
+
+from snuba.request.request_settings import HTTPRequestSettings
+from snuba.datasets.schemas import MandatoryCondition
+from snuba.query.conditions import ConditionFunctions, binary_condition
+from snuba.query.expressions import Column, Literal
+
+
+test_data = [
+    (
+        "table1",
+        [
+            MandatoryCondition(
+                ["deleted", "=", "0"],
+                binary_condition(
+                    None,
+                    ConditionFunctions.EQ,
+                    Column(None, None, "deleted"),
+                    Literal(None, 0),
+                ),
+            )
+        ],
+    ),
+    (
+        "table2",
+        [
+            MandatoryCondition(
+                ["time", "=", "1"],
+                binary_condition(
+                    None,
+                    ConditionFunctions.EQ,
+                    Column(None, None, "time"),
+                    Literal(None, 0),
+                ),
+            ),
+            MandatoryCondition(
+                ["time2", "=", "2"],
+                binary_condition(
+                    None,
+                    ConditionFunctions.EQ,
+                    Column(None, None, "time2"),
+                    Literal(None, 0),
+                ),
+            ),
+        ],
+    ),
+]
+
+
+@pytest.mark.parametrize("table, mand_condition", test_data)
+def test_mand_condition(table: str, mand_condition: List[MandatoryCondition]) -> None:
+
+    body = {
+        "conditions": [
+            ["d", "=", "1"],
+            ["c", "=", "3"],
+            ["a", "=", "1"],
+            ["b", "=", "2"],
+        ],
+    }
+
+    # cols = ColumnSet([("col", String())])
+
+    cols = None
+    consistent = True
+
+    query = Query(body, TableSource(table, cols, mand_condition, ["c1"]),)
```

Comment:
Now. we have two query representations coexisting in the system. The legacy one(body)and the ast one. We should test both, so you should build the query with both a legacy and ast representation.

There are two ways to do this:
1) just pass the ast conditions to the Query initializer. It is the `condition` parameter and it takes an object of type Expression. This is a way for building a condition: https://github.com/getsentry/snuba/blob/master/tests/query/processors/test_prewhere.py#L77-L93. Though we have a few helper functions to make it more concise here: https://github.com/getsentry/snuba/blob/882be95ba0d462a29759a49b2e9aad0c1ce111a9/snuba/query/conditions.py#L112.
In order to create an AND you would create a FunctionCall, with name `and` and two parameters for the two operands. In order to cretate an equality you would create a FunctionCall with name `equal` and one column on the left hand side and a literal on the right hand side. By composing these two you can build the entire tree for your conditions at line 62 to 65.

2) The alternative is to parse the body of the query, but you would need also to instantiate a dataset. I would suggest to stick to option 1 at this time so you get to familiarize yourself with the AST as well.